balancer-js/src/modules/liquidity-managment/migrations.integrations.spec.ts

```diff
@@ -28,296 +29,361 @@ describe('Migrations', function () {
-      const from = metaStable;
-      const to = from;
+    context('user approved the relayer', () => {
+      beforeEach(async () => {
+        await reset('https://rpc.ankr.com/eth', provider, 16950000);
+        signer = await impersonateAccount(address, provider);
+
+        // approve relayer
+        await vault
+          .connect(signer)
+          .setRelayerApproval(address, relayerAddress, true);
+      });
+
+      context('Metastable to Metastable', () => {
+        const from = metaStable;
+        const to = from;
+
+        describe('bptHodler', () => {
+          before(() => {
+            address = '0x21ac89788d52070D23B8EaCEcBD3Dc544178DC60';
+          });
+
+          it('joins a new pool with an limit', async () => {
+            const balance = (
+              await ERC20__factory.connect(from.address, signer).balanceOf(
+                address
+              )
+            ).toString();
+            const peek = await migrations.pool2pool({
+              user: address,
+              from: from.id,
+              to: to.id,
+              balance,
+            });
+            const peekResult = await signer.call({ ...peek, gasLimit: 8e6 });
+            const expectedBptOut = Migrations.getMinBptOut(peekResult);
+
+            const txParams = await migrations.pool2pool({
+              user: address,
+              from: from.id,
+              to: to.id,
+              balance,
+              minBptOut: expectedBptOut,
+            });
+
+            await (await signer.sendTransaction(txParams)).wait();
+
+            const balanceAfter = (
+              await ERC20__factory.connect(to.address, signer).balanceOf(
+                address
+              )
+            ).toString();
+
+            expect(balanceAfter).to.be.eq(expectedBptOut);
+          });
+        });
+
+        describe('staked bpt', () => {
+          before(() => {
+            address = '0xe8343fd029561289CF7359175EE84DA121817C71';
+          });
+
+          it('should build a migration using exit / join and stake tokens in the gauge', async () => {
+            const gauge = (await gaugesRepository.findBy(
+              'poolId',
+              from.id
+            )) as {
+              id: string;
+            };
+            const balance = (
+              await ERC20__factory.connect(gauge.id, signer).balanceOf(address)
+            ).toString();
+
+            const peek = await migrations.pool2poolWithGauges({
+              user: address,
+              from: from.id,
+              to: to.id,
+              balance,
+            });
+            const peekResult = await signer.call({ ...peek, gasLimit: 8e6 });
+            const expectedBptOut = Migrations.getMinBptOut(peekResult);
+
+            const txParams = await migrations.pool2poolWithGauges({
+              user: address,
+              from: from.id,
+              to: to.id,
+              balance,
+              minBptOut: expectedBptOut,
```

Comment:
I've noticed you renamed `expectedBptOut` to `minBptOut` a few times this PR - but there are still a few others that would benefit from the same refactor.

Reply:
hmm.. in fact it should be the other way around:
* `expectedBptOut` is the amount expected from the transaction
* `minBptOut` would be the expected amount including slippage

makes sense? i might need to change the names again

Reply:
These names make sense to me 👍 
What I didn't get is: why are you assigning one to the other, since they are different things?
Is that because on the test you're considering zero slippage?

Reply:
Yes, exactly - for the test i'm expecting 0 slippage from the query amount.